src/App.php

```diff
@@ -1071,25 +1084,24 @@ protected function outputResponse(string $data, array $headers): void
             }
         }
 
-        if ($lateError === null && count($headersNew) > 0 && headers_sent() && !$isCli) {
+        if ($lateError === null && count($headersNew) > 0 && !empty($this->response->getHeaders()) && !$isCli) {
```

Comment:
`headers_sent()` is correct here, it does not check if headers has been built, it checks if some unexpected output has been flushed and thus additional headers can not be flushed, therefore names also "lateError" (which can not be even rendered as we can not tell the browser the content-type will be html.

Reply:
correct, checking via `header_sent` bypass everything and raise an error for unexcepted output